dotCMS/src/main/java/com/dotcms/cms/login/LoginServiceAPIImpl.java

```diff
@@ -0,0 +1,537 @@
+import com.liferay.portal.model.User;
+import com.liferay.portal.util.PropsUtil;
+import com.liferay.portal.util.WebKeys;
+import com.liferay.util.Validator;
+
+/**
+ * Default implementation
+ */
+final class LoginServiceAPIImpl implements LoginServiceAPI {
+
+
+    private static final long serialVersionUID = 1L;
+    private final Log log = LogFactory.getLog(LoginServiceAPI.class);
+    private final UserWebAPI userWebAPI;
+    private final JsonWebTokenUtils jsonWebTokenUtils;
+    private final HttpServletRequestThreadLocal httpServletRequestThreadLocal;
+    private final UserAPI userAPI;
+    private final boolean authByEmail;
+    private final User systemUser;
+    private final String preAuthenticator;
+
+    @VisibleForTesting
+    public LoginServiceAPIImpl(final ApiProvider apiProvider,
+                    final JsonWebTokenUtils jsonWebTokenUtils,
+                    final HttpServletRequestThreadLocal httpServletRequestThreadLocal,
+                    final UserAPI userAPI, boolean authByEmail, String preAuthenticator) {
+
+        this.userWebAPI = apiProvider.userWebAPI();
+        this.jsonWebTokenUtils = jsonWebTokenUtils;
+        this.httpServletRequestThreadLocal = httpServletRequestThreadLocal;
+        this.userAPI = userAPI;
+        this.authByEmail = authByEmail;
+        this.systemUser = APILocator.systemUser();
+        this.preAuthenticator = preAuthenticator;
+        this._listeners.add(new LDAPLoginEventListener());
+
+    }
+
+    @VisibleForTesting
+    public LoginServiceAPIImpl(final ApiProvider apiProvider,
+                    final JsonWebTokenUtils jsonWebTokenUtils,
+                    final HttpServletRequestThreadLocal httpServletRequestThreadLocal,
+                    final UserAPI userAPI) {
+        this(apiProvider, jsonWebTokenUtils, httpServletRequestThreadLocal, userAPI,
+                        (Company.AUTH_TYPE_EA).equals(
+                                        PublicCompanyFactory.getDefaultCompany().getAuthType()),
+                        PropsUtil.get("auth.pipeline.pre"));
+
+    }
+
+    public LoginServiceAPIImpl() {
+        this(new ApiProvider(), JsonWebTokenUtils.getInstance(),
+                        HttpServletRequestThreadLocal.INSTANCE, APILocator.getUserAPI());
+    }
+
+
+
+    @Override
+    public void doLogout(final HttpServletRequest req, final HttpServletResponse res) {
+
+
+        LoginEvent event = new LoginEvent(this, LoginEvents.PRE_LOGOUT, null, null, req, res);
+        for (LoginEventListener listener : _listeners) {
+            listener.loginEventReceived(event);
+        }
+        
+        _doLogoutInternal(req, res);
+        
+        
+        event = new LoginEvent(this, LoginEvents.POST_LOGOUT, null, null, req, res);
+        for (LoginEventListener listener : _listeners) {
+            listener.loginEventReceived(event);
+        }
+    }
+
+
+
+    private void _doLogoutInternal(final HttpServletRequest req, final HttpServletResponse res) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)